### PR TITLE
Add column group header template

### LINKED ISSUE #3344: Grouped header templating
### Description  
Add templating to the Group header of Multi-column headers.

### PR DESCRIPTION
**Pull request by DiyanDimitrov**
Closes #3344

### Additional information (check all that apply):
 - [ ] Bug fix
 - [x] New functionality
 - [ ] Documentation
 - [ ] Demos
 - [ ] CI/CD

### Checklist:
 - [x] All relevant tags have been applied to this PR
 - [x] This PR includes unit tests covering all the new code
 - [ ] This PR includes API docs for newly added methods/properties
 - [ ] This PR includes `feature/README.MD` updates for the feature docs
 - [ ] This PR includes general feature table updates in the root `README.MD`
 - [x] This PR includes `CHANGELOG.MD` updates for newly added functionality
 - [ ] This PR contains breaking changes
 - [ ] This PR includes `ng update` migrations for the breaking changes
 - [ ] This PR includes behavioral changes and the feature specification has been updated with them

### CODE REVIEW

File: projects/igniteui-angular/src/lib/grids/grid/column-group.spec.ts [hunk 1]
```diff
@@ -1368,6 +1369,40 @@ describe('IgxGrid - multi-column headers', () => {
...
+
+        locationColGroup.headerTemplate = headerTemplate;
+        genInfoColGroup.headerTemplate = headerTemplate;
+        fixture.detectChanges();
+
+        const headerSpans: DebugElement[] = fixture.debugElement.queryAll(By.css('.dynamic-col-group-template'));
+        expect(headerSpans.length).toBe(2);
+        headerSpans.forEach(headerSpan => {
+            expect(headerSpan.nativeElement.textContent).toMatch('Dynamic column group template');
+        });
```
Comment (gedinakova) [inline projects/igniteui-angular/src/lib/grids/grid/column-group.spec.ts]: Maybe we can check the template can be successfully removed as well?

Comment (DiyanDimitrov) [reply projects/igniteui-angular/src/lib/grids/grid/column-group.spec.ts]: Done.

**Comment by gedinakova** [comment]
@DiyanDimitrov Collapsing/ expanding the group columns in IE is extremely slow.

**Comment by DiyanDimitrov** [comment]
> @DiyanDimitrov Collapsing/ expanding the group columns in IE is extremely slow.

That's a limitation of the current implementation of the column hiding. The grid sizes are recalculated for each column.hidden change.

**Pull request merged**